cmd/local-gadget/trace/bind.go

```diff
@@ -0,0 +1,76 @@
+// Copyright 2022 The Inspektor Gadget authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package trace
+
+import (
+	"fmt"
+	"math"
+
+	"github.com/cilium/ebpf"
+	"github.com/spf13/cobra"
+
+	commontrace "github.com/kinvolk/inspektor-gadget/cmd/common/trace"
+	commonutils "github.com/kinvolk/inspektor-gadget/cmd/common/utils"
+	"github.com/kinvolk/inspektor-gadget/cmd/local-gadget/utils"
+	"github.com/kinvolk/inspektor-gadget/pkg/gadget-collection/gadgets/trace"
+	"github.com/kinvolk/inspektor-gadget/pkg/gadgets"
+	"github.com/kinvolk/inspektor-gadget/pkg/gadgets/trace/bind/tracer"
+	"github.com/kinvolk/inspektor-gadget/pkg/gadgets/trace/bind/types"
+)
+
+func newBindCmd() *cobra.Command {
+	var commonFlags utils.CommonFlags
+	var flags commontrace.BindFlags
+
+	runCmd := func(cmd *cobra.Command, args []string) error {
+		parsedTargetPorts := make([]uint16, len(flags.TargetPorts))
+		for _, port := range flags.TargetPorts {
```

Comment:
Yeah, that's true. Trying to avoid duplicated code on `kubectl-gadget` and `local-gadget` made all this more complex 🙁. I am planning to finish integrating all the gadgets on `local-gadget` and then, with all of them aligned, try to find a better way to organize the code.

For instance, I would like to check if there is a way to follow your suggestion [here](https://github.com/kinvolk/inspektor-gadget/pull/896#issuecomment-1235317127), and divide the CLI code into three sub-modules with these tasks:

1. Defining CLI commands and flags.
2. Transform CLI flags and env variables into a configuration struct.
3. Take the configuration struct and interact with the server side. 

Where (1) and (2) are shared by `kubectl-gadget` and `local-gadget`, while (3) differs. I need to look for some projects that are doing something similar to get some inspiration. If you have some, please let me know.

Reply:
I have a toy repo I created as an example for this: https://github.com/invidian/golang-cli-testing-example. Maybe that gives you some inspiration. I'm afraid you won't find so many examples of this in the wild.

We also did a similar thing in https://github.com/newrelic/newrelic-infra-operator/blob/main/internal/cli/cli.go.

I also have some extreme private examples which are not yet open source unfortunately :smile:

Reply:
Thanks! Let me take a look!